data/json/materials.json

```diff
@@ -2584,14 +2584,20 @@
     "id": "charcoal",
     "name": "Charcoal",
     "copy-from": "wood",
-    "fuel_data": { "energy": "1600 kJ" }
+    "fuel_data": { "energy": "6240 kJ" },
+    "//": "physical density for porous charcoal sits at about 0.208 kg/liter, while energy density per kg is slightly lower than pure anthracite coal at 30 mJ/kg.",
```

Comment:
http://ukrfuel.com/news-physical-properties-of-charcoal-22.html this was one such source that pointed me towards keeping the pre existing .208 in game but it could definitely change especially to keep a 1:5 ratio of coal:charcoal

Reply:
Okay, firstly please remove the word "porous", because "porous charcoal"/"porous carbon" can be referring to activated charcoal, which is not what we are dealing with here. That threw me off.

>  the pre existing .208 [kg/liter]

It was (10×11g)/250ml = 0.44kg/L, so how is .208 "pre existing"?

> keep a 1:5 ratio of coal:charcoal

Ratio between what property of the two items? Neither mass nor volume were in the ratio 1:5, so you've lost me here.
Also, if the goal is to make charcoal and coal 1:1 on energy, then we should not care about any other preexisting ratios; all that matters is setting equivalent energy, then calculating all other stats so that they are sane values that correspond to the energy amount.

Reply:
it's currently 1:8, but "forcing" it to be 1:5 by changing charcoal density to like .3 g/cm3. I may have misremembered the part about "pre existing" it's been months and I've forgotten to an extent what was pre existing before my PR and what was pre existing before I changed it in the process of my PR.

Reply:
Ok, I think I figured out what the root of the problem is.

My initial comment was indeed a reaction to the 1:8 required volume of coal vs charcoal to deliver the same amount of energy, which intuitively just seemed too much -- and I thought it might be due to using the density of the wrong thing (activated carbon instead of charcoal).

But perhaps it is the difference between bulk density vs "true" density.  
If we have a bucket full of gravel or coal or charcoal, bulk density is the weight of that stuff divided by the volume capacity of the bucket. It includes all the empty space in between the individual pieces. For the purposes of the game, this is the better stat to use when we are trying to determine the weight/volume properties that one unit of an item should have, since this is what determines how much space it will take up in inventory/container
True density is dividing the weight of one solid block of the material by its volume. That number is the one to use in `materials.json` for the purposes of density sanity checks -- it should never be exceeded.

Now, I still don't know where 0.208 comes from -- this specific figure is not explicitly in the given source -- and it certainly isn't "preexisting" -- but it is within reasonable range of charcoal bulk density. (ukrfuel page cited above gives 180 - 220 kg/m³)

The problem is that the volume of coal was derived using true density of 1.5g/cm³, giving 1.5g and 1ml per unit. If we used [bulk density for anthracite coal](https://www.engineeringtoolbox.com/classification-coal-d_164.html), 1.5g of coal would have a volume of somewhere between 1.615ml to 1.875ml. That would mean that instead of 1:8 we'd have something in the range of 1:4.27 to 1:4.95. Pretty huge difference!

Reply:
I'll try to take another look at this also. Raw density vs bulk density (since coal is not uniformly compressed) does matter

Reply:
OK, coal adjusted to match bulk density and being roughly 5:1 kj per liter compared to coal again, which does make for nicer easier math. Charcoal usage also adjusted elsewhere to account for it losing mass from 8ml/1.66g to 5ml/1.04g.

Reply:
@cake-pie let me know if anything still looks wrong, I'm pretty sure I applied all of your requests here.